include/up-cpp/communication/Publisher.h

```diff
@@ -57,7 +57,7 @@ struct Publisher {
 
 private:
 	std::shared_ptr<transport::UTransport> transport_;
-	datamodel::builder::UMessageBuilder publish_builder_;
+	std::shared_ptr<datamodel::builder::UMessageBuilder> publish_builder_;
```

Comment:
Why is this a shared_ptr? Couldn't it just be a regular member?
```suggestion
	datamodel::builder::UMessageBuilder publish_builder_;
```

_note: I'm going to assume this change is made for all following comments_

Reply:
done